examples/tutorials/snapshot.md

```diff
@@ -0,0 +1,611 @@
+write a snapshot file and then read it to test the assertion. If it is the first
+time you are running the test a do not already have a snapshot, add the
+`--update` flag:
+
+```bash
+deno test --allow-read --allow-write -- --update
+```
+
+If you already have a snapshot file, you can run the test with:
+
+```bash
+deno test --allow-read
+```
+
+The test will compare the current output of the object against the stored
+snapshot. If they match, the test passes; if they differ, the test fails.
+
+The snapshot file will look like this:
+
+```ts title="__snapshots__/example_test.ts.snap"
+export const snapshot = {};
+
+snapshot[`isSnapshotMatch 1`] = `
+{
+  example: 123,
+  hello: "world!",
+}
+`;
+```
+
+You can edit your test to change the `hello` string to `"everyone!"` and run the
+test again with `deno test --allow-read`. This time the `assertSnapshot`
+function will throw an `AssertionError`, causing the test to fail because the
+snapshot created during the test does not match the one in the snapshot file.
+
+## Updating snapshots
+
+When adding new snapshot assertions to your test suite, or when intentionally
+making changes which cause your snapshots to fail, you can update your snapshots
+by running the snapshot tests in update mode. Tests can be run in update mode by
+passing the `--update` or `-u` flag as an argument when running the test. When
+this flag is passed, then any snapshots which do not match will be updated.
+
+```bash
+deno test --allow-read --allow-write -- --update
+```
+
+:::note
+
+New snapshots will only be created when the `--update` flag is present.
+
+:::
+
+## Permissions
+
+When running snapshot tests, the `--allow-read` permission must be enabled, or
+else any calls to `assertSnapshot` will fail due to insufficient permissions.
+Additionally, when updating snapshots, the `--allow-write` permission must be
+enabled, as this is required in order to update snapshot files.
+
+The assertSnapshot function will only attempt to read from and write to snapshot
+files. As such, the allow list for `--allow-read` and `--allow-write` can be
+limited to only include existing snapshot files, if desired.
+
+## Version Control
+
+Snapshot testing works best when changes to snapshot files are committed
+alongside other code changes. This allows for changes to reference snapshots to
+be reviewed along side the code changes that caused them, and ensures that when
+others pull your changes, their tests will pass without needing to update
+snapshots locally.
+
+## Options
+
+The `assertSnapshot` function can be called with an `options` object which
+offers greater flexibility and enables some non standard use cases:
+
+```ts
+import { assertSnapshot } from "jsr:@std/testing/snapshot";
+
+Deno.test("isSnapshotMatch", async function (t): Promise<void> {
+  const a = {
+    hello: "world!",
+    example: 123,
+  };
+  await assertSnapshot(t, a, {/*custom options go here*/});
```

Comment:
Not a blocking issue, but I'm curious... what sort of options are we talking about here? Is there an way to illustrate that example?

Reply:
...and then I kept reading :)